### ISSUE TITLE
Google Spreadsheets worksheet by title cleanup/refactor

### ISSUE THREAD

**Issue by guidopetri** [getredash/redash#6409]
In #5334 , I left a few comments about future work on the "fetch worksheet by title" functionality for the google spreadsheets query runner. We should address those, namely:
- don't use mixed-type variables
- allow for non-quoted worksheet titles (single word)
- verify 0/1 indexing of worksheet numbers, and add comment if it differs

**Comment by guidopetri** [comment]
@TanyaKansal absolutely! Feel free to submit a PR and tag me in it when it's ready for review :)

**Comment by immortalcodes** [comment]
@guidopetri  I didn't quite understand the 3rd point , btw for other 2 please have a look https://github.com/getredash/redash/compare/master...immortalcodes:redash:google-sheet-singleword-title

**Comment by immortalcodes** [comment]
@guidopetri  I didn't quite understand the 3rd point , btw for other 2 please have a look https://github.com/getredash/redash/compare/master...immortalcodes:redash:google-sheet-singleword-title

**Comment by justinclift** [comment]
Note - removed duplicated comment above. :wink:

**Comment by DhanushA1307** [comment]
Hi. I am new and would like to contribute to this issue @guidopetri
